Store password, if requested

xabgesagtx opened the pull request:
* enable the user to optionally store the password
* show warning to the user
* store password in settings
* allow user to delete the stored password (without signing out)

Fixes #33

czoller commented:
The "Delete Password" setting doesn't appear and disappear as expected.
For example, when first started Kodi "Delete Password" is in the settings although I have never stored one (maybe this was a one time problem).
But more critical, when signed out, it is still there although I cannot use the password to sign in again. If it is not attended for the password to be usable for manual sign-in then it should be deleted when signed out.

xabgesagtx commented:
> For example, when first started Kodi "Delete Password" is in the settings although I have never stored one (maybe this was a one time problem).

I encountered the same issue that the delete button appeared even though I didn't use the new feature yet. I assumed that this is a remainder of an old version, where the password was stored always.
I deleted the settings.xml to simulate a fresh install and the "DELETE PASSWORD" option was gone. After signing in without saving the password the option was still not visible. So, I think this might not be an issue for any user who newly installs the addon without any prior versions of the addon ever being installed.

> But more critical, when signed out, it is still there although I cannot use the password to sign in again. If it is not attended for the password to be usable for manual sign-in then it should be deleted when signed out.

Good point. The password gets now removed on sign out as well.

czoller reviewed (approved):
Looks good now
